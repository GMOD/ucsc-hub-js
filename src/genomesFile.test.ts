// @ts-nocheck
import fs from 'fs'
import GenomesFile from './genomesFile'

describe('GenomesFile', () => {
  it('reads a basic genomes.txt file', () => {
    const input = fs.readFileSync('test/basic.genomes.txt', 'utf8')
    const genomesFile = new GenomesFile(input)
    expect(genomesFile).toMatchSnapshot()
    expect(genomesFile.toString()).toEqual(input)
  })

  it("throws if the file doesn't start with a genomes entry", () =>
    expect(
      () => new GenomesFile('assembly hg19\ntrackDb hg19/trackDb.txt'),
    ).toThrow(/file must begin with a line like/))

  it('throws if a genome entry is missing a required field', () =>
    expect(
      () =>
        new GenomesFile(
          'genome hg19\ntrackDb hg19/trackDb.txt\n\ngenome hg38\norganism Homo sapiens',
        ),
    ).toThrow(/file entry .* is missing required entry:/))
})
