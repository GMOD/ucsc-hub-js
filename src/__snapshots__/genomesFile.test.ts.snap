// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenomesFile reads a basic genomes.txt file 1`] = `
Map {
  "hg18" => Map {
    "genome" => "hg18",
    "trackDb" => "hg18/trackDb.txt",
  },
  "hg19" => Map {
    "genome" => "hg19",
    "trackDb" => "hg19/trackDb.txt",
  },
  "newOrg1" => Map {
    "genome" => "newOrg1",
    "trackDb" => "newOrg1/trackDb.txt",
    "twoBitPath" => "newOrg1/newOrg1.2bit",
    "groups" => "newOrg1/groups.txt",
    "description" => "Big Foot V4",
    "organism" => "BigFoot",
    "defaultPos" => "chr21:33031596-33033258",
    "orderKey" => "4800",
    "scientificName" => "Biggus Footus",
    "htmlPath" => "newOrg1/description.html",
  },
}
`;
